### PR TITLE
Fix/tweak context parsing

### PR DESCRIPTION
**Pull request by mahatoankitkumar**
## Problem
For number in context, it would consider it as an integer as whole, even if it was an array

## Solution
Fixed the parsing for number

### CODE REVIEW

File: crates/frontend/src/utils.rs [hunk 1]
```diff
@@ -319,17 +319,22 @@ pub fn get_config_value(
...
-                f64::from_str(val)
-                    .ok()
-                    .and_then(|num| Number::from_f64(num).map(Value::Number))
-                    .ok_or_else(|| {
-                        "Invalid decimal format or precision issue".to_string()
-                    })
-            }),
+        Some(ConfigValueType::Number) => {
+            let parsed_value: Value = serde_json::from_str(val)
+                .map_err(|_| "Invalid number or number array format".to_string())?;
```
Comment (sauraww) [inline crates/frontend/src/utils.rs:324]: why do we need a number array format inside Number ?
Can you give an example or a screenshot ?

**Review by sauraww — CHANGES_REQUESTED** [review]

File: crates/frontend/src/utils.rs (hunk 1 above)
Comment (mahatoankitkumar) [reply crates/frontend/src/utils.rs:324]: for `in` case, if you give [1,2,3], it will try to convert it as number as a whole.

Comment (sauraww) [reply crates/frontend/src/utils.rs:324]: Can we separate this logic based on the operator? We need similar validation for booleans and strings as well. Also, the code looks messy. We should keep the existing logic, but move the array logic to a different function for better organization and handling.

Comment (sauraww) [reply crates/frontend/src/utils.rs:324]: @mahatoankitkumar

**Review by sauraww — CHANGES_REQUESTED** [review]

File: crates/frontend/src/utils.rs (hunk 1 above)
Comment (mahatoankitkumar) [reply crates/frontend/src/utils.rs:324]: @sauraww Why would we need similar validation for booleans and strings?

Comment (sauraww) [reply crates/frontend/src/utils.rs:324]: There can be strings in the array right like list of clientIds, we can have a use case like that

**Review by sauraww — CHANGES_REQUESTED** [review]

**Pull request closed without merging**